src/amo/components/MastHead.js

```diff
@@ -34,10 +33,12 @@ export class MastHeadBase extends React.Component {
 
     return (
       <div className="MastHead">
-        <div id="tabzilla">
-          <a href="https://www.mozilla.org">Mozilla</a>
+        <div className="MastHead-top-row">
+          <div id="tabzilla">
+            <a href="https://www.mozilla.org">Mozilla</a>
+          </div>
+          <AuthenticateButton className="MastHead-auth-button" size="small" location={location} />
```

Comment:
I thought there was a `withRouter` HOC you could pass to `compose`? Would that work?